.github/actions/matrix-fanin/action.yml

```diff
@@ -0,0 +1,19 @@
+name: 'Matrix Fanin'
+description: 'Collect all JSON files from the matrix and combine them into a single file (matrix-fanin.json).'
+inputs:
+  json-file-prefix:
+    description: 'The prefix of the JSON files to combine.'
+    required: true
+runs:
+  using: 'composite'
+  steps:
+    - name: Download artifacts
+      uses: 'actions/download-artifact@v4'
+      with:
+        merge-multiple: true
+        pattern: ${{ inputs.json-file-prefix }}*
```

Comment:
matrix-output/

Reply:
Artifacts aren't stored in directories. When uploaded they are stored in the top-level with the name provided, and we can't specify a directory since that isn't a valid name. I did a lot of research into this and I can't find a way of doing it the way you describe, if you think it is possible, feel free to build off of this.